### PR TITLE
[openssl] Change repo to github and update to 3.0.4

### PR DESCRIPTION
**Pull request by JackBoosY**
This PR will:
- Change repo to github
- Update openssl version to 3.0.4

Override https://github.com/microsoft/vcpkg/pull/25430 if the test passed.

### CODE REVIEW

**Comment by talregev** [comment]
I am closing my PR. if you want to go back only for switch to GitHub, I will open it again. Thank you for the fix and upgrade openssl.

**Review by LilyWangLL — APPROVED** [review]

File: ports/openssl/windows/flags.patch [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
+diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
+index 66bc81d..2364633 100644
+--- a/Configurations/10-main.conf
++++ b/Configurations/10-main.conf
+@@ -1302,7 +1302,7 @@ my %targets = (
+         inherit_from     => [ "BASE_Windows" ],
+         template         => 1,
+         CC               => "cl",
+-        CPP              => '"$(CC)" /EP /C',
++        CPP              => '$(CC) /EP /C',
```
[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/openssl/windows/flags.patch:10]: This just fix the command error:
```
	D:\downloads\tools\jom\jom-1.1.3\jom.exe /j1 depend
	D:\downloads\tools\jom\jom-1.1.3\jom.exe /j1 _build_libs
	cmd /C "set "ASM="nasm"" & "D:\downloads\tools\perl\5.32.1.1\perl\bin\perl.exe" "crypto\aes\asm\aes-586.pl" "win32n" -I"." -I"include" -I"providers\common\include" -I"providers\implementations\include" /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP  -MD -O2 -Oi -Gy -DNDEBUG   -FS -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"D:\\packages\\openssl_x86-windows\"" -D"ENGINESDIR=\"D:\\packages\\openssl_x86-windows\\lib\\engines-3\"" -D"MODULESDIR=\"D:\\packages\\openssl_x86-windows\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"  -DAES_ASM -DCMLL_ASM -DDES_ASM -DECP_NISTZ256_ASM -DGHASH_ASM -DMD5_ASM -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_PART_WORDS -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_USE_APPLINK -DPADLOCK_ASM -DPOLY1305_ASM -DRMD160_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DVPAES_ASM -DWHIRLPOOL_ASM  crypto\aes\aes-586.asm.S"
	""C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.32.31326/bin/Hostx64/x86/cl.exe"" /EP /C  -I"." -I"include" -I"providers\common\include" -I"providers\implementations\include" /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP  -MD -O2 -Oi -Gy -DNDEBUG   -FS -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"D:\\packages\\openssl_x86-windows\"" -D"ENGINESDIR=\"D:\\packages\\openssl_x86-windows\\lib\\engines-3\"" -D"MODULESDIR=\"D:\\packages\\openssl_x86-windows\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"   -DAES_ASM -DCMLL_ASM -DDES_ASM -DECP_NISTZ256_ASM -DGHASH_ASM -DMD5_ASM -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_PART_WORDS -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_USE_APPLINK -DPADLOCK_ASM -DPOLY1305_ASM -DRMD160_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DVPAES_ASM -DWHIRLPOOL_ASM crypto\aes\aes-586.asm.S > crypto\aes\aes-586.asm.i

'""C:' is not recognized as an internal or external command,
```

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/openssl/windows/flags.patch:10]: Is this the right fix? I think that `cmd` actually needs `\` path separators for the command. Removing the quotes won't help but introduce other problems.

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/openssl/windows/flags.patch:10]: .. but I see the double quotes.

[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/openssl/windows/flags.patch:10]: @dg0yt You can see `""C:` is come from `""C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.32.31326/bin/Hostx64/x86/cl.exe""`.

[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/openssl/windows/flags.patch:10]: The reason for the build failure should be that `CC` has included a double quote once and added it again in the `CPP`, which causes the compiler to think `""C:` is a separate part, and uses `/` as a delimiter to separate different command parts.

**Pull request merged**